x-pack/plugins/cloud/public/plugin.tsx

```diff
@@ -230,109 +242,158 @@ export class CloudPlugin implements Plugin<CloudSetup> {
     return user?.roles.includes('superuser') ?? true;
   }
 
-  private async setupFullstory({
-    basePath,
+  /**
+   * If the right config is provided, register the FullStory shipper to the analytics client.
+   * @param analytics Core's Analytics service's setup contract.
+   * @param basePath Core's http.basePath helper.
+   * @private
+   */
+  private async setupFullStory({ analytics, basePath }: SetupFullStoryDeps) {
+    const { enabled, org_id: fullStoryOrgId } = this.config.full_story;
+    if (!enabled || !fullStoryOrgId) {
+      return; // do not load any FullStory code in the browser if not enabled
+    }
+
+    // Keep this import async so that we do not load any FullStory code into the browser when it is disabled.
+    const { FullStoryShipper } = await import('@elastic/analytics');
```

Comment:
Given we know that the `@elastic/analytics` will at least always be imported by `core`, should we move it to the shared libs?

Reply:
I can create a follow-up PR to test what's the gain in bundle size 👍